packages/dds/tree/src/test/feature-libraries/treeCursorUtils.spec.ts

```diff
@@ -0,0 +1,264 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { compareFieldUpPaths, compareUpPaths, FieldUpPath, UpPath } from "../../core";
+// Allow importing from this specific file which is being tested:
+// eslint-disable-next-line import/no-internal-modules
+import { PrefixedPath, prefixFieldPath, prefixPath } from "../../feature-libraries/treeCursorUtils";
+import { brand } from "../../util";
+
+describe("treeCursorUtils", () => {
+    const root: UpPath = {
+        parent: undefined,
+        parentField: brand("x"),
+        parentIndex: 5,
+    };
+    const child: UpPath = {
+        parent: root,
+        parentField: brand("z"),
+        parentIndex: 10,
+    };
+    describe("prefixPath", () => {
+        it("not wrapped cases", () => {
+            assert.equal(prefixPath(undefined, undefined), undefined);
+            assert.equal(prefixPath({}, undefined), undefined);
+            assert.equal(prefixPath(undefined, root), root);
+            assert.equal(prefixPath({}, root), root);
+            assert.equal(
+                prefixPath({ indexOffset: 1, rootFieldOverride: brand("y") }, undefined),
+                undefined,
+            );
+        });
+
+        it("wrapped root", () => {
+            assert(
+                compareUpPaths(prefixPath({ indexOffset: 1 }, root), {
+                    parent: undefined,
+                    parentField: brand("x"),
+                    parentIndex: 6,
+                }),
+            );
+            assert(
+                compareUpPaths(prefixPath({ rootFieldOverride: brand("y") }, root), {
+                    parent: undefined,
+                    parentField: brand("y"),
+                    parentIndex: 5,
+                }),
+            );
+            assert(
+                compareUpPaths(prefixPath({ parent: root }, root), {
```

Comment:
This case is confusing to me. What does making the root path's parent be itself actually mean?

Reply:
Like the two cases above, this is just resting one of the fields in PathRootPrefix, in this case `parent`. `root` is simply the simplest UpPath thats in scope to use as the parent, and also the simplest UpPath we can apply the prefix to.